packages/components/tooltip/src/tooltip.tsx

```diff
@@ -244,21 +258,33 @@ const Tooltip = (props: TTooltipProps) => {
         onBlur(event);
       }
 
-      if (closeAfter) {
+      if (closeAfter && state === 'opened') {
         leaveTimer.current = setTimeout(() => {
-          handleClose(event);
+          const tooltipElement = popperInstance?.popper.querySelector(
```

Comment:
When the close timeout is done, we can't just remove the tooltip but start the exit transition. We do this by setting the state to `exiting`.
We also register an event to be notified when the transition is finished to actually remove the tooltip from the DOM.